app/upgrades/v22/upgrades.go

```diff
@@ -4,20 +4,49 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
 	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	stakeibckeeper "github.com/Stride-Labs/stride/v21/x/stakeibc/keeper"
 )
 
-const (
+var (
 	UpgradeName = "v22"
+
+	CosmosHubChainId = "cosmoshub-4"
+	OsmosisChainId   = "osmosis-1"
+	DydxChainId      = "dydx-mainnet-1"
+
+	MaxMessagesPerIcaByHost = map[string]uint64{
```

Comment:
DYDX/Osmosis have higher blockly gas limits than CH (infinite and 300M), so think these limits should be ok for both

For CH, the gas to stake/unstake is somewhat unpredictable because it scales with additional ICS tokens.

Trying to track down pStake/QS ICA accounts to see how many validators they're delegating to

Reply:
111 for hub
56 for dydx 
86 for osmo